agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/quickpulse/QuickPulseDataCollector.java

```diff
@@ -121,7 +126,8 @@ private CountAndDuration(long count, long duration) {
 
     final AtomicLong rddsAndDuations = new AtomicLong(0);
     final AtomicInteger unsuccessfulRdds = new AtomicInteger(0);
-    final List<QuickPulseDocument> documentList = new ArrayList<>();
+    final ArrayBlockingQueue<QuickPulseDocument> documentList =
+        new ArrayBlockingQueue<>(1000, true);
```

Comment:
I think `ArrayBlockingQueue` is a bit heavy-weight for our needs (designed for concurrency).

I think `List` with a simple size check before adding should be enough. this is just a fallback safety mechanism now that you have implemented checking for quickpulse `on` before adding